rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java

```diff
@@ -1710,10 +1738,38 @@ private <J2 extends J> JRightPadded<J2> convert(Tree t, Function<Tree, Space> su
         J2 j = convert(t);
         @SuppressWarnings("ConstantConditions") JRightPadded<J2> rightPadded = j == null ? null :
                 new JRightPadded<>(j, suffix.apply(t), Markers.EMPTY);
-        cursor(max(endPos(t), cursor)); // if there is a non-empty suffix, the cursor may have already moved past it
+        int idx = findFirstNonWhitespaceChar(rightPadded.getAfter().getWhitespace());
+        if (idx >= 0) {
+            rightPadded = (JRightPadded<J2>) JRightPadded.build(getErroneous(List.of(rightPadded)));
+        }
+        if (endPos(t) == cursor && rightPadded.getElement() instanceof J.Erroneous) {
+            cursor++;
```

Comment:
This part is unclear to me. Why advance the cursor a single character? Can we add a comment explaining this? Is there a specific test case which would demonstrate this?

Reply:
I'd have to dig into this to recall why this was done. Hopefully a test would fail if I comment this out. Will get back on this.

Reply:
Added a comment